internal/tools/cloudflare/dns.go

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	cf "github.com/cloudflare/cloudflare-go"
+	"github.com/pkg/errors"
+
+	"github.com/sirupsen/logrus"
+)
+
+const defaultTimeout = 30 * time.Second
+
+// Cloudflarer interface that holds Cloudflare functions
+type Cloudflarer interface {
+	ZoneIDByName(zoneName string) (string, error)
+	DNSRecords(ctx context.Context, zoneID string, rr cf.DNSRecord) ([]cf.DNSRecord, error)
+	CreateDNSRecord(ctx context.Context, zoneID string, rr cf.DNSRecord) (*cf.DNSRecordResponse, error)
+	DeleteDNSRecord(ctx context.Context, zoneID, recordID string) error
+}
+
+// Client struct
+type Client struct {
+	cfClient Cloudflarer
+}
+
+// NewClientWithToken creates a new client that can be used to run the other functions.
+func NewClientWithToken(client Cloudflarer) *Client {
+	return &Client{
+		cfClient: client,
+	}
+}
+
+func (c *Client) getZoneID(zoneName string) (zoneID string, err error) {
+	zoneID, err = c.cfClient.ZoneIDByName(zoneName)
+	if err != nil {
+		return "", err
+	}
+
+	return zoneID, err
```

Comment:
Lets be explicit when returning `nil`.
```suggestion
	return zoneID, nil
```